lib/jack_tokenizer.rb

```diff
@@ -11,12 +11,21 @@ def initialize(io)
   LETTERS = ("a".."z").to_a + ("A".."Z").to_a
 
   def has_more_tokens?
-    while WHITESPACES.include?(@input[@index])
-      @index += 1
-    end
+    loop do
```

Comment:
The body of this loop is getting pretty large. Do you think it’d be worth extracting some private helper methods (e.g. `#skip_whitespace`, `#skip_comment`) to make it shorter and easier to understand? Not essential, just an option.